mat-process-api.Tests/V1/Controllers/ProcessDataImagesControllerTests.cs

```diff
@@ -175,5 +182,329 @@ public void given_an_exception_thrown_check_that_the_controller_throws_correct_e
+
+            //act
+            _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
+
+            //assert
+            _mockUsecase.Verify(u => u.ExecuteGet(It.IsAny<GetProcessImageRequest>()), Times.Once);
+        }
+
+        [Test]
+        public void given_a_valid_request_when_GetProcessImage_controller_method_is_called_then_it_calls_usecase_with_correct_data()
+        {
+            //arrange
+            GetProcessImageRequest request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
+            _mockGetValidator.Setup(x => x.Validate(It.IsAny<GetProcessImageRequest>())).Returns(new FV.ValidationResult()); //setup validator to return a no error validation result
+
+            //act
+            _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
+
+            //assert
+            _mockUsecase.Verify(u => u.ExecuteGet(It.Is<GetProcessImageRequest>(obj =>
+                obj.processRef == request.processRef &&
+                obj.imageId == request.imageId
+                )), Times.Once);
+        }
+
+        [Test]
+        public void given_a_request_when_GetProcessImage_controller_method_is_called_then_it_calls_the_validator_with_that_request_object()
+        {
+            //arrange
+            var request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
+           
+            _mockGetValidator.Setup(x => x.Validate(It.IsAny<GetProcessImageRequest>())).Returns(new FV.ValidationResult()); //setup validator to return a no error validation result
+
+            //act
+            _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
+
+            //assert
+            _mockGetValidator.Verify(v => v.Validate(It.Is<GetProcessImageRequest>(obj => obj.fileExtension == request.fileExtension)), Times.Once);
+            _mockGetValidator.Verify(v => v.Validate(It.Is<GetProcessImageRequest>(obj => obj.imageId == request.imageId)), Times.Once);
+            _mockGetValidator.Verify(v => v.Validate(It.Is<GetProcessImageRequest>(obj => obj.processRef == request.processRef)), Times.Once);
+            _mockGetValidator.Verify(v => v.Validate(It.Is<GetProcessImageRequest>(obj => obj.processType == request.processType)), Times.Once);
+        }
+
+        [Test]
+        public void when_GetProcessImage_controller_method_is_called_then_it_calls_the_logger()
+        {
+            //arrange
+            var request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
+            _mockGetValidator.Setup(l => l.Validate(It.IsAny<GetProcessImageRequest>())).Returns(new FV.ValidationResult()); //setup validator to return a no error validation result
+
+            //act
+            _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
```

Comment:
**This is repeated comment, but same issue in different place:**

> Related to comment '<1>' . For this test the contents of the request don't matter, I think this controller method call should be supplied with nulls to indicate that to avoid confusion about the test. Also it makes test harder to read and follow.
> 
> Like that:
> `var controllerResponse = _processImageController.GetProcessImage(null, null, null ,null);`